synapse/lib/ingest.py

```diff
@@ -182,6 +182,104 @@ def iterdata(fd,**opts):
 
     fd.close()
 
+class IngestApi:
+    '''
+    An API mixin which may be used to wrap a cortex and send along ingest data.
+    '''
+    def __init__(self, core):
+        self._gest_core = core
+        self._gest_cache = {}
+
+        self._gest_core.on('tufo:del:syn:ingest', self._onDelSynIngest)
+        self._gest_core.on('tufo:add:syn:ingest', self._onAddSynIngest)
+        self._gest_core.on('tufo:set:syn:ingest:text', self._onAddSynIngest)
+
+        for node in self._gest_core.getTufosByProp('syn:ingest'):
+            self._addDefFromTufo(node)
+
+    def _onAddSynIngest(self, mesg):
+        tufo = mesg[1].get('tufo')
+        return self._addDefFromTufo(tufo)
+
+    def _addDefFromTufo(self, tufo):
+
+        name = tufo[1].get('syn:ingest')
+        if name == None:
+            logger.warning('_addDefFromTufo syn:ingest == None')
+            return
+
+        text = tufo[1].get('syn:ingest:text')
+        if text == None:
+            logger.warning('_addDefFromTufo syn:ingest:text == None')
+            return
+
+        try:
+
+            info = json.loads(text)
+            self._gest_cache[name] = Ingest(info)
+
+        except Exception as e:
+            logger.exception('_addDefFromTufo json loading failed')
+            return
+
+    def _onDelSynIngest(self, mesg):
```

Comment:
There is no corresponding public API to remove a named ingest from the cortex.  Adding a delGestDef function would allow complete removal of a ingest definition.